### PR TITLE
👷 Adding Github actions analysis workflow

### PR DESCRIPTION
**Pull request by devbyaccident**
#### What scientific question is your analysis addressing?

N/A, this is a new feature to add a workflow using Github actions

#### What was your approach?

Because the analysis needed to be done inside a container built within this repo, this workflow is built to run as a [private Github action](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action#example-using-a-private-action=) that uses the **action.yml** file at the root of this repository to define a custom entrypoint script that accepts arguments, then send each analysis command as an argument to that entrypoint.

Since these will run on the same Github runner as a part of the same job, the container is built on the first step in the job, then used when running the additional analysis commands.

#### What GitHub issue does your pull request address?

Connects [#63](https://github.com/PediatricOpenTargets/ticket-tracker/issues/63)

#### Which areas should receive a particularly close look?

Make sure that the [OpenPBTA testing data](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/62a1cc0b2a8db8a582e1106c5b64de62ef3d96d9/.github/workflows/run-analysis.yml#L19) in addition to the container size is under 14 GB or else the workflow will exceed the [hardware limitations of the Github actions runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources=).

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

File: .github/workflows/run-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Run Analysis
+
+# Controls when the action will run.
+on:
+  workflow_dispatch:
```
Comment (devbyaccident) [inline .github/workflows/run-analysis.yml:5]: I tend to leave `workflow_dispatch` in my actions so they can be run outside of pushing a new commit. If this is an issue, please let me know and I'll remove it.

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: "Run OpenPedCan Analysis"
+description: "Run analysis workflow"
+runs:
+  using: "docker"
+  image: "Dockerfile"
+  entrypoint: "scripts/run-analysis.sh"
```
Comment (devbyaccident) [inline action.yml:6]: Uses the script at `scripts/run-analysis.sh` as an [entrypoint for the action](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsentrypoint=) without modifying the dockerfile. Good find on this flag @logstar!

File: .github/workflows/run-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Build container and download data
+        uses: ./
+        id: download-data
+        with:
+          args: OPENPBTA_URL=https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/data OPENPBTA_RELEASE=testing bash download-data.sh
+
+      - name: Run 01-create-interaction-plots in container
+        uses: ./
+        id: create-interaction-plots
```
Comment (devbyaccident) [inline .github/workflows/run-analysis.yml:23]: ID's here need to be unique and not contain numbers. I've made a note of this in the readme as well.

File: scripts/run-analysis.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+sh -c "$*"
```
Comment (devbyaccident) [inline scripts/run-analysis.sh:3]: This is really simple. It'll just take any arguments defined the current step in [.github/workflows/run-analysis.yml](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/183/files#diff-6f60327802f48f21c83edf1c43fdc1b2bcc69434ab8e14bfd7a5e7bc75ce6e7a) and run them as shell scripts inside the container.

File: README.md [hunk 5]
```diff
@@ -425,7 +425,7 @@ bash scripts/download-ci-files.sh
 
 Running this will change the symlinks in `data` to point to the files in `data/testing`.
 
-#### Adding Analyses to CI
+#### Adding Analyses to Circle CI
 
 For an analysis to be run in CI, it must be added to the Circle CI configuration file, [`.circleci/config.yml`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/.circleci/config.yml).
```
Comment (logstar) [inline README.md]:
```suggestion
For an analysis to be run in CI, it must be added to the Circle CI configuration file, [`.circleci/config.yml`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/.circleci/config.yml).
```

File: README.md [hunk 6]
```diff
@@ -450,6 +450,41 @@ If you wanted to add running an Rscript called `cluster-samples.R` that was in a
 
 This would run the `cluster-samples.R` on the subset files that are specifically designed to be used for CI.
 
+#### Adding Analyses to Github Actions workflow
+
+For an analysis to be run in a Github Actions workflow, it must be added to [`.github/run-analysis.yml`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/dev/.github/run-analysis.yml). Here is an example of a step in that workflow:
```
Comment (logstar) [inline README.md]:
```suggestion
For an analysis to be run in a Github Actions workflow, it must be added to [`.github/run-analysis.yml`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/.github/workflows/run-analysis.yml). Here is an example of a step in that workflow:
```

File: README.md [hunk 7]
```diff
@@ -450,6 +450,41 @@ If you wanted to add running an Rscript called `cluster-samples.R` that was in a
...
+
+```yaml
+      - name: Run RUN-telomerase-activity-prediction in container
+        uses: ./
+        id: RUN-telomerase-activity-prediction
+        with:
+          args: bash analyses/telomerase-activity-prediction/RUN-telomerase-activity-prediction.sh
+```
+
+The `uses:` tag specifies for the step to run using the action defined in the [`action.yml`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/dev/action.yml) file.
```
Comment (logstar) [inline README.md]:
```suggestion
The `uses:` tag specifies for the step to run using the action defined in the [`action.yml`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/action.yml) file.
```

File: README.md [hunk 8]
```diff
@@ -450,6 +450,41 @@ If you wanted to add running an Rscript called `cluster-samples.R` that was in a
...
+```yaml
+name: "Run OpenPedCan Analysis"
+description: "Run analysis workflow"
+runs:
+  using: "docker"
+  image: "Dockerfile"
+  entrypoint: "scripts/run-analysis.sh"
+```
+
+In this file, the workflow is run in the docker container defined in [`Dockerfile`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/master/Dockerfile) using [`scripts/run-analysis.sh`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/dev/scripts/run-analysis.sh) as an entrypoint script. Any commands defined under the `with: args:` tags in [`.github/run-analysis.yml`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/dev/.github/run-analysis.yml) will run as commands inside of the docker container when the action is run.
```
Comment (logstar) [inline README.md]:
```suggestion
In this file, the workflow is run in the docker container defined in [`Dockerfile`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/Dockerfile) using [`scripts/run-analysis.sh`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/scripts/run-analysis.sh) as an entrypoint script. Any commands defined under the `with: args:` tags in [`.github/run-analysis.yml`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/.github/run-analysis.yml) will run as commands inside of the docker container when the action is run.
```

File: README.md [hunk 9]
```diff
@@ -450,6 +450,41 @@ If you wanted to add running an Rscript called `cluster-samples.R` that was in a
...
+
+To add a new analysis job, take the template below and value each missing prompt, then add it to [`.github/run-analysis.yml`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/dev/.github/run-analysis.yml)
```
Comment (logstar) [inline README.md]:
```suggestion
To add a new analysis job, take the template below and value each missing prompt, then add it to [`.github/run-analysis.yml`](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/.github/run-analysis.yml)
```

File: README.md (hunk 5 above)
Comment (devbyaccident) [reply README.md:430]: > I added several comments and corrected some URLs in README.md.

Thanks! I'll commit these now!

**Comment by devbyaccident** [comment]
> I added several comments and corrected some URLs in README.md.

Thanks @logstar! I'll commit these now!

**Review by logstar — APPROVED** [review]

**Pull request merged**